### PR TITLE
Adding test fixes for test suites failing with segrep enabled

### PR DESCRIPTION
**Pull request by Poojita-Raj**
### Description
Fixes flaky failing tests with segment replication enabled with added test fixes. 

### Issues Resolved
[List any issues this PR will resolve]

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [ ] Commits are signed per the DCO using --signoff
- [ ] Commit changes are listed out in CHANGELOG.md file (See: [Changelog](../blob/main/CONTRIBUTING.md#changelog))

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: server/src/internalClusterTest/java/org/opensearch/indices/mapping/ConcurrentDynamicTemplateIT.java [hunk 1]
```diff
@@ -94,9 +94,10 @@ public void onFailure(Exception e) {
             latch.await();
             assertThat(throwable, emptyIterable());
             refresh();
-            assertHitCount(client().prepareSearch("test").setQuery(QueryBuilders.matchQuery(fieldName, "test-user")).get(), numDocs);
-            assertHitCount(client().prepareSearch("test").setQuery(QueryBuilders.matchQuery(fieldName, "test user")).get(), 0);
-
+            assertBusy(() -> {
+                assertHitCount(client().prepareSearch("test").setQuery(QueryBuilders.matchQuery(fieldName, "test-user")).get(), numDocs);
+                assertHitCount(client().prepareSearch("test").setQuery(QueryBuilders.matchQuery(fieldName, "test user")).get(), 0);
+            });
```
Comment (dreamer-89) [inline server/src/internalClusterTest/java/org/opensearch/indices/mapping/ConcurrentDynamicTemplateIT.java:100]: nit: By default, assertBusy waits for `10s`. We can add a longer wait time e.g. 1 minute owning due to nature of segrep where replica refresh may take time.

File: server/src/internalClusterTest/java/org/opensearch/search/pit/PitMultiNodeIT.java [hunk 2]
```diff
@@ -58,7 +58,7 @@
 /**
  * Multi node integration tests for PIT creation and search operation with PIT ID.
  */
-@OpenSearchIntegTestCase.ClusterScope(scope = OpenSearchIntegTestCase.Scope.TEST, numDataNodes = 2)
+@OpenSearchIntegTestCase.ClusterScope(scope = OpenSearchIntegTestCase.Scope.TEST, minNumDataNodes = 2)
```
Comment (dreamer-89) [inline server/src/internalClusterTest/java/org/opensearch/search/pit/PitMultiNodeIT.java:61]: Why this change is needed ?

File: server/src/internalClusterTest/java/org/opensearch/search/sort/SimpleSortIT.java [hunk 3]
```diff
@@ -198,45 +198,49 @@ public void testSimpleSorts() throws Exception {
         }
         refresh();
 
-        // STRING script
         int size = 1 + random.nextInt(10);
+        final SearchResponse[] searchResponse = new SearchResponse[1];
+        // STRING script
```
Comment (dreamer-89) [inline server/src/internalClusterTest/java/org/opensearch/search/sort/SimpleSortIT.java:203]: unintended

File: server/src/internalClusterTest/java/org/opensearch/search/sort/SimpleSortIT.java [hunk 4]
```diff
@@ -198,45 +198,49 @@ public void testSimpleSorts() throws Exception {
...
+                .get();
+            assertHitCount(searchResponse[0], 10);
+        });
 
-        assertHitCount(searchResponse, 10);
-        assertThat(searchResponse.getHits().getHits().length, equalTo(size));
+        assertThat(searchResponse[0].getHits().getHits().length, equalTo(size));
         for (int i = 0; i < size; i++) {
-            SearchHit searchHit = searchResponse.getHits().getAt(i);
+            SearchHit searchHit = searchResponse[0].getHits().getAt(i);
```
Comment (dreamer-89) [inline server/src/internalClusterTest/java/org/opensearch/search/sort/SimpleSortIT.java:219]: Why this change is needed ?

File: server/src/main/java/org/opensearch/index/IndexSettings.java [hunk 5]
```diff
@@ -752,13 +751,13 @@ public IndexSettings(final IndexMetadata indexMetadata, final Settings nodeSetti
...
-//        } else {
-//            replicationType = IndexMetadata.INDEX_REPLICATION_TYPE_SETTING.get(settings);
-//        }
+        // if (FeatureFlags.isEnabled(FeatureFlags.SEGMENT_REPLICATION_EXPERIMENTAL)
+        // && indexMetadata.isSystem() == false
+        // && settings.get(IndexMetadata.SETTING_REPLICATION_TYPE) == null) {
+        // replicationType = IndicesService.CLUSTER_REPLICATION_TYPE_SETTING.get(settings);
+        // } else {
+        // replicationType = IndexMetadata.INDEX_REPLICATION_TYPE_SETTING.get(settings);
+        // }
```
Comment (dreamer-89) [inline server/src/main/java/org/opensearch/index/IndexSettings.java:760]: Can you clean this up ?

File: server/src/internalClusterTest/java/org/opensearch/search/pit/PitMultiNodeIT.java (hunk 2 above)
Comment (Poojita-Raj) [reply server/src/internalClusterTest/java/org/opensearch/search/pit/PitMultiNodeIT.java:61]: I'll remove it

File: server/src/internalClusterTest/java/org/opensearch/search/sort/SimpleSortIT.java (hunk 4 above)
Comment (Poojita-Raj) [reply server/src/internalClusterTest/java/org/opensearch/search/sort/SimpleSortIT.java:219]: To allow for searchResponse to be effectively final within the assertBusy scope. 

An alternative would be to move this loop into the assertBusy scope as well and redefine the next searchResponse outside the scope.

Comment (dreamer-89) [reply server/src/internalClusterTest/java/org/opensearch/search/sort/SimpleSortIT.java:219]: Yeah, we can move entire fetch response and verification under single assertBusy.

**Review by Rishikesh1159 — APPROVED** [review]

**Review by dreamer-89 — APPROVED** [review]
